platform/o.n.swing.tabcontrol/src/org/netbeans/swing/popupswitcher/SwitcherTable.java

```diff
@@ -161,7 +161,7 @@ public Component prepareRenderer(
         if (ren instanceof JLabel) {
             // #199007: Swing HTML renderer does a poor job of truncating long labels
             JLabel prototype = (JLabel) ren;
-            lbl = HtmlRenderer.createLabel();
+            lbl = (JLabel) HtmlRenderer.createRenderer();
```

Comment:
I think this could be improved if lbl would be declared to be a  `JComponent`. Most calls will still work, and the cases where that is not the case (calls to setIcon/setText) are already null protected and could just as well be protected with `if lbl instanceof JLabel)` and only cast at that point.

Reply:
Ignore this - it is not cleaner, the return of `HtmlRenderer#createRenderer` does not allow this.